joy/src/joy_node_linux.cpp

```diff
@@ -1,102 +1,121 @@
-    double interval = now - lastDiagTime_;
-    if (open_)
-      stat.summary(0, "OK");
-    else
-      stat.summary(2, "Joystick not open.");
-    
-    stat.add("topic", pub_.getTopic());
-    stat.add("device", joy_dev_);
-    stat.add("dead zone", deadzone_);
-    stat.add("autorepeat rate (Hz)", autorepeat_rate_);
-    stat.add("coalesce interval (s)", coalesce_interval_);
-    stat.add("recent joystick event rate (Hz)", event_count_ / interval);
-    stat.add("recent publication rate (Hz)", pub_count_ / interval);
-    stat.add("subscribers", pub_.getNumSubscribers());
-    event_count_ = 0;
-    pub_count_ = 0;
-    lastDiagTime_ = now;
-  }
-  
+
+  // // TODO(mikaelarguedas) commenting out diagnostic logic for now
+  // int event_count_;
+  // int pub_count_;
+  // double lastDiagTime_;
+  // 
+  // diagnostic_updater::Updater diagnostic_;
+  // 
+  // ///\brief Publishes diagnostics and status
+  // void diagnostics(diagnostic_updater::DiagnosticStatusWrapper& stat)
+  // {
+  //   double now = rclcpp::Time::now().toSec();
+  //   double interval = now - lastDiagTime_;
+  //   if (open_)
+  //     stat.summary(0, "OK");
+  //   else
+  //     stat.summary(2, "Joystick not open.");
+  //   
+  //   stat.add("topic", pub_.getTopic());
+  //   stat.add("device", joy_dev_);
+  //   stat.add("dead zone", deadzone_);
+  //   stat.add("autorepeat rate (Hz)", autorepeat_rate_);
+  //   stat.add("coalesce interval (s)", coalesce_interval_);
+  //   stat.add("recent joystick event rate (Hz)", event_count_ / interval);
+  //   stat.add("recent publication rate (Hz)", pub_count_ / interval);
+  //   stat.add("subscribers", pub_.getNumSubscribers());
+  //   event_count_ = 0;
+  //   pub_count_ = 0;
+  //   lastDiagTime_ = now;
+  // }
+  //
+
 public:
-  Joystick() : nh_(), diagnostic_()
+  Joystick()
   {}
   
   ///\brief Opens joystick port, reads from port and publishes while node is active
   int main(int argc, char **argv)
   {
-    diagnostic_.add("Joystick Driver Status", this, &Joystick::diagnostics);
-    diagnostic_.setHardwareID("none");
+    // diagnostic_.add("Joystick Driver Status", this, &Joystick::diagnostics);
+    // diagnostic_.setHardwareID("none");
+    (void)argc;
+    (void)argv;
+
+    auto node = std::make_shared<rclcpp::Node>("joy_node");
+
+    auto pub = node->create_publisher<sensor_msgs::msg::Joy>("joy");
 
     // Parameters
-    ros::NodeHandle nh_param("~");
-    pub_ = nh_.advertise<sensor_msgs::Joy>("joy", 1);
-    nh_param.param<std::string>("dev", joy_dev_, "/dev/input/js0");
-    nh_param.param<double>("deadzone", deadzone_, 0.05);
-    nh_param.param<double>("autorepeat_rate", autorepeat_rate_, 0);
-    nh_param.param<double>("coalesce_interval", coalesce_interval_, 0.001);
+    node->get_parameter_or("dev", joy_dev_, std::string("/dev/input/js0"));
+    node->get_parameter_or("deadzone", deadzone_, 0.05);
+    node->get_parameter_or("autorepeat_rate", autorepeat_rate_, static_cast<double>(0));
```

Comment:
I couldn't quite get things working with this node, with the robot jerking around instead of smoothly following the joystick commands.  After a bit of debugging, setting the autorepeat_rate to 0 is the reason why that is.  In the old version of the code I did, I had autorepeat_rate set to 20, which is how the ROS1 cartographer startup code configures it.  If I set this to 20 in this node, then things start working again, so I think that is what we should do here.